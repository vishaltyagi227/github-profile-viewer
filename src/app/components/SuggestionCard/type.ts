export interface ISuggestionCard {
  userName: string;
  imageUrl: string;
  closeBtn?: boolean;
  onClose?: (userName: string) => void;
}
