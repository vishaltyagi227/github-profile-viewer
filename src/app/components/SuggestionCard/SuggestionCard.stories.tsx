import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Story, Meta } from '@storybook/react';
import SuggestionCard from '@components/SuggestionCard';
import { ISuggestionCard } from '@components/SuggestionCard/type';

export default {
  title: 'Components/SuggestionCard',
  component: SuggestionCard,
} as Meta;

const Template: Story<ISuggestionCard> = ({ ...args }) => (
  <BrowserRouter>
    <SuggestionCard {...args} />
  </BrowserRouter>
);

export const Basic = Template.bind({});

Basic.argTypes = {
  closeBtn: { control: { disable: true } },
};
Basic.args = {
  userName: 'vishal',
  imageUrl: 'imageUrl.png',
};

export const WithCloseIcon = Template.bind({});

WithCloseIcon.args = {
  ...Basic.args,
  closeBtn: true,
};
