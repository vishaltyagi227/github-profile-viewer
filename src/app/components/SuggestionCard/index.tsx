import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import {
  Card,
  Avatar,
  CardContent,
  CardActions,
  Typography,
  IconButton,
} from '@mui/material';
import { Close } from '@mui/icons-material';

import { ROUTE, VARIANT, COMPONENT } from '@constant/index';
import Link from '@components/Link';
import useStyles from './style';
import { ISuggestionCard } from './type';

const SuggestionCard = ({
  userName,
  imageUrl,
  closeBtn,
  onClose,
}: ISuggestionCard) => {
  const classes = useStyles();

  return (
    <Card className={classes.suggestionCard}>
      <Avatar src={imageUrl} alt={userName} className={classes.cardAvatar} />
      <CardContent>
        <Typography gutterBottom variant={VARIANT.h5} component={COMPONENT.div}>
          {userName}
        </Typography>
      </CardContent>
      <CardActions className={classes.cardActions}>
        <Link
          component={RouterLink}
          to={`${ROUTE.ArbitraryUser}${userName}`}
          className={classes.cardLink}
        >
          View
        </Link>
        {closeBtn && (
          <IconButton
            className={classes.closeBtn}
            onClick={() => onClose && onClose(userName)}
          >
            <Close />
          </IconButton>
        )}
      </CardActions>
    </Card>
  );
};
SuggestionCard.defaultProps = {
  closeBtn: false,
};
export default SuggestionCard;
