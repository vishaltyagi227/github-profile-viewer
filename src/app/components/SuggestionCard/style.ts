import { createStyles, makeStyles } from '@mui/styles';
import { Theme } from '@mui/material/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    suggestionCard: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      paddingInline: theme.spacing(5),
      paddingBlock: theme.spacing(2),
      [theme.breakpoints.up(theme.breakpoints.values.sm)]: {
        flexDirection: 'row',
      },
    },
    cardAvatar: {
      width: theme.pxToRem(80),
      height: theme.pxToRem(80),
    },
    cardActions: {
      flexGrow: 1,
      display: 'flex',
      justifyContent: 'flex-end',
    },
    cardLink: {
      color: theme.customPallete.secondary.light,
      textDecoration: 'none',
      '&:hover': {
        color: theme.customPallete.primary.main,
      },
    },
    closeBtn: {
      marginLeft: theme.spacing(1),
    },
  })
);

export default useStyles;
