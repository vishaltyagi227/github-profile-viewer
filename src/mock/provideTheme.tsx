import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
