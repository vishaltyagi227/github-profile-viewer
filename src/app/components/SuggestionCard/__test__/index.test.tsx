import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import SuggestionCard from '../index';

const details = {
  userName: 'vishal',
  avatarUrl: 'https://dummyurl.com/img',
};

describe('<SuggestionCard />', () => {
  let container: RenderResult;

  afterEach(cleanup);

  it('Should render correctly', () => {
    container = render(
      provideTheme(
        <BrowserRouter>
          <SuggestionCard
            userName={details.userName}
            imageUrl={details.avatarUrl}
            closeBtn
            onClose={jest.fn()}
          />
        </BrowserRouter>
      )
    );
    expect(container.getByAltText(details.userName)).toBeInTheDocument();
    expect(container.getByText(details.userName)).toBeInTheDocument();
    expect(container.getByText(/View/i)).toBeInTheDocument();
    expect(container.getByRole('button')).toBeInTheDocument();
  });
});

describe('<SuggestionCard /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with Snapshot', () => {
    const tree = renderer
      .create(
        provideTheme(
          <BrowserRouter>
            <SuggestionCard
              userName={details.userName}
              imageUrl={details.avatarUrl}
              closeBtn
              onClose={jest.fn()}
            />
          </BrowserRouter>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
