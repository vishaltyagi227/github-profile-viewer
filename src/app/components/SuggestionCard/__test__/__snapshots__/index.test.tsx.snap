// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionCard /> Snapshot Should match with Snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-suggestionCard-6 css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiAvatar-root MuiAvatar-circular makeStyles-cardAvatar-7 css-1wlk0hk-MuiAvatar-root"
  >
    <img
      alt="vishal"
      className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
      src="https://dummyurl.com/img"
    />
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-do8s7i-MuiTypography-root"
    >
      vishal
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing makeStyles-cardActions-8 css-1t6e9jv-MuiCardActions-root"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways makeStyles-cardLink-9 css-1iwtne7-MuiTypography-root-MuiLink-root"
      href="/user/vishal"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      View
    </a>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium makeStyles-closeBtn-10 css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
